plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
+		if checkIntegerOverFlow(totalPriority, config.Weight*MaxPriority) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * MaxPriority
+	}
 	return configs, nil
 }
 
+// checkIntegerOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkIntegerOverFlow(totalPriority int, weight int) bool {
```

Comment:
ok, sorry, now that I better understand whats going on here, im going to suggest you call this `addPrioritySafely `, 

- Check weight*Maxpriority multiplies w/o overflow.
- then check that totalPriority increments w/o overflow.
- Return error if fails.
- Otherwise return the new added on value of totalPriority

That will be super easy to understand.  Sorry about changing my mind :).